sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1016,6 +1016,8 @@ class FilterPushdownSuite extends PlanTest {
     val correctAnswer = x.where("x.a".attr === 5).join(y.where("y.a".attr === 5),
         condition = Some("x.a".attr === Rand(10) && "y.b".attr === 5))
 
-    comparePlans(Optimize.execute(originalQuery.analyze), correctAnswer.analyze)
```

Comment:
Filter is allowed to have nondeterministic condition.

For Join, actually the rule to handle nondeterministic condition is unnecessary, as join condition is always deterministic, we can remove these logic in the optimizer rule and its corresponding tests.